src/scripts/main.js

```diff
@@ -1,3 +1,323 @@
+        newColumn[j - 1] = 0;
+        newColumn[j] = newColumn[j] * 2;
+        score += newColumn[j];
+      }
+    }
+
+    newColumn = shiftRowLeft(newColumn).reverse();
+
+    for (let j = 0; j < newColumn.length; j++) {
+      board[j][i] = newColumn[j];
+    }
+  }
+}
+
+function changeAdditionalClassCell(element, newAddClass) {
+  element.classList = '';
+  element.classList.add('field-cell');
+
+  if (newAddClass) {
+    element.classList.add(newAddClass);
+  }
+}
+
+function ifWin() {
+  for (const row of board) {
+    for (const cell of row) {
+      if (cell === 2048) {
+        messageWin.classList.remove('hidden');
+        start.textContent = 'Start';
+        start.classList.remove('restart');
+        start.classList.add('start');
+        started = false;
+        messageStart.classList.remove('hidden');
+      }
+    }
+  }
+}
+
+function freeCell() {
+  for (const row of board) {
+    for (const cell of row) {
+      if (cell === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function ifLose() {
+  let lose = true;
+  const temp = [...board];
+  const tempProbabilityCount = probabilityCount;
+  const tempScore = score;
+
+  moveLeft();
+
+  if (freeCell()) {
+    lose = false;
+    board = [...temp];
+    probabilityCount = tempProbabilityCount;
+    score = tempScore;
+
+    return lose;
+  }
+
+  moveUp();
+
+  if (freeCell()) {
+    lose = false;
+    board = [...temp];
+    probabilityCount = tempProbabilityCount;
+    score = tempScore;
+
+    return lose;
+  }
+
+  board = [...temp];
+  probabilityCount = tempProbabilityCount;
+  score = tempScore;
+
+  return lose;
+}
+
+function display() {
+  for (let i = 0; i < board.length; i++) {
+    const cells = fieldRowAll[i].querySelectorAll('.field-cell');
+
+    for (let j = 0; j < board[i].length; j++) {
+      if (board[i][j] !== 0) {
+        cells[j].textContent = board[i][j];
+
+        const addClass = 'field-cell--' + board[i][j];
+
+        changeAdditionalClassCell(cells[j], addClass);
+      } else {
+        cells[j].textContent = '';
+        changeAdditionalClassCell(cells[j]);
+      }
+    }
+  }
+
+  scoreDisplay.textContent = score;
+}
+
+function startGame() {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+  probabilityCount = 0;
+
+  newCell();
+  newCell();
+  display();
+}
+
+start.addEventListener('click', ourEvent => {
+  startGame();
+  started = true;
+
+  if (ourEvent.target.matches('.start')) {
+    start.textContent = 'Restart';
+    start.classList.remove('start');
+    start.classList.add('restart');
+    messageStart.classList.add('hidden');
+    messageWin.classList.add('hidden');
+    messageLose.classList.add('hidden');
+  }
+});
+
+document.addEventListener('keyup', e => {
+  if (started === false) {
+    return;
+  }
+
+  switch (e.key) {
+    case 'ArrowRight':
+      moveRight();
+      display();
+      ifWin();
+      newCell();
+      display();
```

Comment:
it would be nice to move these functions outside the switch to avoid duplicating